scripts/diagnostic_gen.py

```diff
@@ -28,17 +27,24 @@ def main():
     parser.add_argument("--srcDir", help="Source directory to search for usages")
     parser.add_argument("--incDir", help="Include directory to search for usages")
     parser.add_argument("--docs", action="store_true")
+    parser.add_argument("--diagnostics", help="path to diagnostics file")
     parser.add_argument("--slangBin")
     args = parser.parse_args()
 
-    ourdir = os.path.dirname(os.path.realpath(__file__))
-    inf = open(os.path.join(ourdir, "diagnostics.txt"))
+    inf = None
+    try:
+        inf = open(args.diagnostics)
+    except Exception:
+        print("failed to open", args.diagnostics)
```

Comment:
This doesn't seem a good use of a catch handler for the exception; you're going to exit anyway, just with a less informative error than you'd get if you let the exception bubble up naturally.

Reply:
Agreed. However, the existing code had a `pass` in the catch block, and that means we'll get an exception later (when we try to write files) if the directory creation failed.

If you're happy having the exception bubble up, I can remove `sys.exit`. But I do think we should handle or throw it here, so that at least the user is pointed to the right location of the error.

Reply:
Yeah, just let the exception bubble up here if it happens.

Reply:
Ack.